### PR TITLE
rmgdft: Update for version 5.0.1

### PR DESCRIPTION
**Pull request by elbriggs**
Updated for rmgdft version 5.0.1 release.

### CODE REVIEW

File: var/spack/repos/builtin/packages/rmgdft/package.py [hunk 1]
```diff
@@ -43,11 +44,15 @@ class Rmgdft(CMakePackage):
     )
 
     # RMGDFT 4.0.0 or later requires compiler support for C++14
-    compiler_warning = "RMGDFT 4.0.0 or later requires a " "compiler with support for C++14"
-    conflicts("%gcc@:4", when="@3.6.0:", msg=compiler_warning)
-    conflicts("%intel@:17", when="@3.6.0:", msg=compiler_warning)
-    conflicts("%pgi@:17", when="@3.6.0:", msg=compiler_warning)
-    conflicts("%llvm@:3.4", when="@3.6.0:", msg=compiler_warning)
+    compiler_warning14 = "RMGDFT 4.0.0 or later requires a " "compiler with support for C++14"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/rmgdft/package.py]:
```suggestion
    compiler_warning14 = "RMGDFT 4.0.0 or later requires a compiler with support for C++14"
```

File: var/spack/repos/builtin/packages/rmgdft/package.py [hunk 2]
```diff
@@ -43,11 +44,15 @@ class Rmgdft(CMakePackage):
...
+    conflicts("%gcc@:4", when="@3.6.0:", msg=compiler_warning14)
+    conflicts("%intel@:17", when="@3.6.0:", msg=compiler_warning14)
+    conflicts("%pgi@:17", when="@3.6.0:", msg=compiler_warning14)
+    conflicts("%llvm@:3.4", when="@3.6.0:", msg=compiler_warning14)
+
+    # RMGDFT 5.0.0 requires C++17 and increase the minimum gcc to 8
+    compiler_warning17 = "RMGDFT 5.0.0 or later requires a " "compiler with support for C++17"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/rmgdft/package.py]:
```suggestion
    compiler_warning17 = "RMGDFT 5.0.0 or later requires a compiler with support for C++17"
```

**Review by tldahlgren — COMMENTED** [review]
Confirmed the new version tag.  Would like to see the "adjacent" quotes for the compiler warnings removed.

**Comment by elbriggs** [comment]
I don't understand why the pipeline failed but somehow this commit was associated with an unrelated merge request.

**Comment by alalazo** [comment]
@spackbot run pipeline

**Review by tldahlgren — APPROVED** [review]
LGTM

**Pull request merged**